server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -349,28 +357,37 @@ private void sendVerifyShardBeforeCloseRequest(final IndexShardRoutingTable shar
     /**
      * Step 3 - Move index states from OPEN to CLOSE in cluster state for indices that are ready for closing.
      */
-    static ClusterState closeRoutingTable(final ClusterState currentState, final Map<Index, AcknowledgedResponse> results) {
+    static ClusterState closeRoutingTable(final ClusterState currentState,
+                                          final ClusterBlock closingBlock,
+                                          final Map<Index, AcknowledgedResponse> results) {
         final MetaData.Builder metadata = MetaData.builder(currentState.metaData());
         final ClusterBlocks.Builder blocks = ClusterBlocks.builder().blocks(currentState.blocks());
         final RoutingTable.Builder routingTable = RoutingTable.builder(currentState.routingTable());
 
         final Set<String> closedIndices = new HashSet<>();
         for (Map.Entry<Index, AcknowledgedResponse> result : results.entrySet()) {
             final Index index = result.getKey();
+            final boolean acknowledged = result.getValue().isAcknowledged();
             try {
                 final IndexMetaData indexMetaData = metadata.getSafe(index);
-                if (indexMetaData.getState() != IndexMetaData.State.CLOSE) {
-                    if (result.getValue().isAcknowledged()) {
-                        logger.debug("closing index {} succeed, removing index routing table", index);
+                if (acknowledged) {
```

Comment:
I tried to make this logic easy to understand but we can maybe do better, any suggestion welcome.

I also think that if we go with always keeping a single block in cluster state we should compute the AcknowledgedResponse by diffing the current state and the updated state.